ci/cppcheck/cppcheck-new-issues.sh

```diff
@@ -13,6 +13,10 @@ rootdir="${scriptdir%/*/*}"
 # Remove the caret lines, we don't need them:
 sed -i '/^\s*\^\s*$/d' "$rootdir/cppcheck_results.txt"
 
+# Sort the list of issues to have a stable list, easier to compare:
+sort "$rootdir/cppcheck_results.txt" > "$rootdir/cppcheck_results_sorted.txt"
+mv "$rootdir/cppcheck_results_sorted.txt" "$rootdir/cppcheck_results.txt"
+
```

Comment:
It would actually make more sense if cppcheck.sh would not create any file but simply output to stdout, and that it was the CI job that creates the output file. I find it pretty annoying that running a tool like that creates a file (which BTW isn't even in gitignore).

With that, you could use a nice pipeline here: `cppcheck.sh | sed | sort > results`.

Reply:
I agree with @arnout - scripts that generate files are less compose-able and are therefore harder to use in other contents. A mid solution is to add an _option_ to output to a file and use stdout as the default for this option.